C/cryptography/Affine_Substitution_Cipher.c

```diff
@@ -0,0 +1,133 @@
+#include<stdio.h>
+#include<string.h>
+#include<ctype.h>
+#include<stdlib.h>
+
+// function declaration
+int CalculateGCD(int);
+int CalculateMultiplicativeInverse(int);
+
+// C program to illustate Affine Cipher
+int main() {
+  int index, index_2, k, gcd, alpha, beta, num_msg[100], numcipher[100], numdecipher[100], decipher[100], alphaInverse;
+  char msg[100], cipher[100];
+  printf("Enter Cipher message : ");
+  scanf("%[^\n]%*c", msg);
+  //converting entered string to Capital letters
+  for (index = 0, index_2 = 0; index < strlen(msg); index++) {
+    if (msg[index] != ' ') {
+      msg[index_2] = toupper(msg[index]);
+      index_2++;
+    } else {
+      msg[index_2] = ' ';
+      index_2++;
+    }
+  }
+  msg[index_2] = '\0';
+  printf("Enter the value of key alpha :");
+  scanf("%d", & alpha);
+  //Checking whether key alpha is in range of(1-25)
+  if (alpha < 1 || alpha > 25) {
+    printf("Alpha should lie in between 1 and 25\nTry again !\n");
+    exit(0);
+  }
+  //  calling CalculateGCD function
+  gcd = CalculateGCD(alpha);
+  if (gcd != 1) {
+    printf("Mandotory condition : gcd(alpha,26)=1 but \n gcd(%d,26)=%d\n Try again !\n", alpha, gcd);
+    exit(0);
+  }
+  printf("Enter the value of key beta : ");
+  scanf("%d", & beta);
+  //Checking whether key beta is in range of(0-25)
+  if (beta < 0 || beta > 25) {
+    printf("Beta value should lie between 0 and 25\n");
+    exit(0);
+  }
+  //Conditions Over
+  //Storing message in terms of ASCII and replacing spaces used  to -20
+  for (index = 0; index < strlen(msg); index++) {
+    if (msg[index] != ' ')
+      num_msg[index] = msg[index] - 'A';
+    else
+      num_msg[index] = -20;
+  }
+  //Ciphering Process
+  /* If numcipher is more than 25 .We need to convert and ensure that lie
+    in between 0 and 25.(indicating Alphabets) A-0,B-1,C-2,.....Y-24,Z-25 */
+  printf("Encrypted Message is :");
+  for (index = 0; index < strlen(msg); index++) {
+    if (num_msg[index] != -20) {
+      numcipher[index] = ((alpha * num_msg[index]) + beta) % 26;
+      printf("%c", (numcipher[index] + 'A'));
+      cipher[index] = numcipher[index] + 'A';
+    }
+    else {
+      printf(" ");
+      cipher[index] = ' ';
+    }
+ }
+   //Deciphering Process
+  printf("\nDecrypted Message is : ");
+  for (index = 0; index < strlen(cipher); index++) {
```

Comment:
Format code according to [C style guide](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/C_Style_guide.md)